mypy/subtypes.py

```diff
@@ -228,9 +228,14 @@ def visit_typeddict_type(self, left: TypedDictType) -> bool:
         elif isinstance(right, TypedDictType):
             if not left.names_are_wider_than(right):
                 return False
-            for (_, l, r) in left.zip(right):
+            for name, l, r in left.zip(right):
                 if not is_equivalent(l, r, self.check_type_parameter):
                     return False
+                # Non-required key is not compatible with a required key since indexing
```

Comment:
Hm, I wonder if there's something to say for making at least one of the directions work here??? It's hard to keep things straight (see my notes on some of the tests).